### DISCUSSION TITLE
Is the Veracruz rustdoc hosted anywhere?

### DISCUSSION THREAD

**Discussion by geky** [veracruz-project/veracruz#74] [Q&A] [answered]
I could just be missing where the docs are hosted, if so let me know (and then this discussion could help guide users yay).

Assuming these aren't already hosted:

One nice thing Rust provides is a pretty rendering of the rustdoc comments. I could generated these locally, but that's one more speed bump for potential users.

If we put Veracruz on crates.io, that would provide automatic hosting on docs.rs, though I can also understand crates.io may not be a good match. For one each crate would get its own site, which for Veracruz's current layout may make it hard to connect between.

Another option is to use GitHub pages, with some work we could automatically render the rustdocs in a subdirectory on https://veracruz-project.github.io/ for example.

**Comment by dominic-mulligan-arm** [comment, marked as answer]
No, not yet.  There's an open issue on the issue tracker regarding adding missing Rustdoc comments (as a soft-rule, any new pull request should now be fully Rustdoc'd too, so that coverage slowly accretes over time).  I think the SDK is fully Rustdoc'd, parts of Mexico City and Chihuahua are too, and Derek's latest Nitro changes were also Rustdoc'd, but the rest of the codebase is not.  At the moment, there's not even a `doc` make target in the top-level Makefile, if I remember correctly, though some second-tier ones do, e.g. `sdk/Makefile` which just puts the generated documents in `target/doc`.

**Reply by geky** [reply]
This is a bit tangential to actually writing the rustdocs. There's already quite a bit of documentation that would benefit from being hosted somewhere publicly readable.

Currently to read the rustdocs you either just read the source (which is a bit difficult with private/public functions mixed together) or generate it locally, which adds a barrier to reading the documentation.

**Reply by dominic-mulligan-arm** [reply]
Yeah, sorry I went a bit off-topic.  I agree that hosting the Rustdoc documentation somewhere (probably the project homepage) is a good idea.  I'll open an issue.

**Reply by dominic-mulligan-arm** [reply]
See issue #76.
